pkg/minikube/machine/cache_images.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machine
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sync/errgroup"
+
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/bootstrapper"
+	"k8s.io/minikube/pkg/minikube/constants"
+
+	"github.com/containers/image/copy"
+	"github.com/containers/image/docker"
+	"github.com/containers/image/docker/archive"
+	"github.com/containers/image/signature"
+	"github.com/containers/image/types"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+const tempLoadDir = "/tmp"
+
+func CacheImagesForBootstrapper(clusterBootstrapper string) error {
+	images, ok := bootstrapper.CachedImagesForBootstrapper[clusterBootstrapper]
+	if !ok {
+		glog.Infoln("Could not find list of images to cache for bootstrapper %s", clusterBootstrapper)
+		return nil
+	}
+
+	if err := CacheImages(images, constants.ImageCacheDir); err != nil {
+		return errors.Wrapf(err, "Caching images for %s", clusterBootstrapper)
+	}
+
+	return nil
+}
+
+func CacheImages(images []string, cacheDir string) error {
```

Comment:
A comment explaining what the resulting cache directory will look like might help.